src/jdk.hotspot.agent/doc/transported_core.html

```diff
@@ -10,101 +10,98 @@ <h1>Debugging transported core dumps</h1>
-Solaris Linkers and Libraries Guide - 817-1984</a>
+On  most platforms, core dumps do not contain text (code) pages.
+Their pages are instead read from the executable and shared objects (or DLLs).
+Therefore it is important to have a matching java executable and shared object
+files on the debugger machine. The best way to guarantee this match is to match the
+debugger machine to that of core dump machine. This means having the same
+OS version and libraries, and also having the same version of the JDK. It also means having
+the OS libraries and JDK installed in the same locations on both machines. However, often
+this isn't an option, and instead you need to copy the libraries and java installation from
+the machine where the core dump was produced to the debugger machine, and you need
+to tell SA where these are all located. That is done differently for each
+OS type, and is described in the following sections.
 </p>
 
-<h3>Solaris SA against transported core dumps</h3>
-
+<h3>Using  transported core dumps on Linux</h3>
 <p>
-With transported core dumps, you may get "rtld_db failures" or
-"libthread_db failures" or SA may just throw some other error
-(hotspot symbol is missing) when opening the core dump. 
-Environment variable <b>LIBSAPROC_DEBUG</b> may be set to any value
-to debug such scenarios. With this env. var set, SA prints many
-messages in standard error which can be useful for further debugging.
-SA on Solaris uses <b>libproc.so</b> library. This library also
-prints debug messages with env. var <b>LIBPROC_DEBUG</b>. But,
-setting LIBSAPROC_DEBUG results in setting LIBPROC_DEBUG as well.
+On Linux, SA parses core and shared library ELF files. But, you
+may still face problems with transported core dumps, because matching shared
+objects may not be in the path(s) specified in core dump file. To
+workaround this, you can define environment variable <b>SA_ALTROOT</b>
+to be the directory where shared libraries are kept. In this directory you should
+copy all the needed JDK libraries, and any other user JNI libraries that were used.
+You should also copy any referenced linux
+libraries from the core dump machine, unless they are identical to what are
+installed on the debugger machine.
 </p>
 <p>
-The best possible way to debug a transported core dump is to match the
-debugger machine to that of core dump machine. i.e., have same Kernel
-and libthread patch level between the machines. mdb (Solaris modular
-debugger) may be used to find the Kernel patch level of core dump
-machine and debugger machine may be brought to the same level.
+ You can also use <b>SA_ALTROOT</b> to specify a path mapping. For example, if you
+ set <b>SA_ALTROOT=/altroot</b>, then <b>/altroot</b> will be prepended to any path found
+ in the core file, and also prepended to any subdir with the root part stripped off. For example,
+ when looking up <b>/usr/lib/libfoo.so</b>, SA will try to find <b>/altroot/usr/lib/libfoo.so</b>,
```

Comment:
Nit: Not sure if it makes sense to say: "SA will first try to find".

Reply:
I decided to keep this as-is since it will try to find all 3, and I felt saying `first...and then...and finally...` was a bit wordy.

Reply:
Okay.